File: 卡码/代码随想录/1/01-数组/C++/03-有序数组的平方/LeetCode0977.cpp
//977. 有序数组的平方
#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
  vector<int> sortedSquares(vector<int>& nums) {
		vector<int> res(nums.size());
		int i=0,j=nums.size()-1,k=j;
		while(i<=j){
			if(pow2(nums[i])>pow2(nums[j])) res[k--]=pow2(nums[i++]);
			else res[k--]=(int)pow2(nums[j--]);
		}
		return res;
  }
  int pow2(int n){
  	return n*n;
  }
};

int main(){
	vector<int> nums={-4,-1,0,3,10};
	Solution* solution=new Solution();
	for(int value:solution->sortedSquares(nums)) printf("%d ",value);
	return 0;
}

File: 卡码/代码随想录/1/01-数组/Go/03-有序数组的平方/LeetCode_0977.go
//977. 有序数组的平方
package main

import "fmt"

func sortedSquares(nums []int) []int {
	res:=make([]int,len(nums))
	k:=len(nums)-1
	for i,j:=0,len(nums)-1;i<=j; {
		if pow2(nums[i])>pow2(nums[j]) {
			res[k]=pow2(nums[i])
			i++
		}else{
			res[k]=pow2(nums[j])
			j--
		}
		k--
	}
	return res
}

func pow2(n int) int{
	return n*n
}

func main(){
	nums:=[]int{-4,-1,0,3,10}
	fmt.Println(sortedSquares(nums))
}
